message/models.py

```diff
@@ -0,0 +1,61 @@
+from django.db import models
+from django.utils import timezone
+from dogowner.models import DogOwner
+from daycare.models import DayCare
+
+
+class AuthorOptions(models.TextChoices):
+    DogOwner = 'O', 'DogOwner'
+    DayCare = 'D', 'DayCare'
+
+
+class Message(models.Model):
+    author = models.CharField(max_length=1,
+                              choices=AuthorOptions.choices)
+    dogowner_id = models.ForeignKey(DogOwner, on_delete=models.CASCADE)
+    daycare_id = models.ForeignKey(DayCare, on_delete=models.CASCADE)
+    text = models.TextField()
+    date = models.DateTimeField(default=timezone.now)
+    message_reply = models.ForeignKey('self', on_delete=models.CASCADE, null=True)
+
+    def __str__(self):
+
+        if(self.author == AuthorOptions.DayCare):
+            res = f"From: {self.daycare_id}"
+            res += f" To: {self.dogowner_id}"
+        else:
+            res = f"From: {self.dogowner_id}"
+            res += f" To: {self.daycare_id}"
+
+        res += f" text: {self.text}"
+        res += f" date: {self.date}"
+        return res
+
+    @staticmethod
+    def create(author, dogowner_id, daycare_id, text):
+        last = Message.get_last_message_between_dogowner_daycare(
+            dogowner_id, daycare_id)
+
+        new_message = Message(author=author,
+                              dogowner_id=dogowner_id,
+                              daycare_id=daycare_id,
+                              text=text,
+                              message_reply=last)
+
+        new_message.save()
+        return new_message
+
+    @staticmethod
+    def get_all_dog_owner_message(user_id):
+        return Message.objects.filter(dogowner_id=user_id)
+
+    @staticmethod
+    def get_all_daycare_message(user_id):
```

Comment:
I dont't get why those methods exists. can you explain how do you see the message feature on the app?

Reply:
removed